src/core/ext/client_config/client_channel.c

```diff
@@ -33,6 +33,7 @@
 
 #include "src/core/ext/client_config/client_channel.h"
```

Comment:
Done.  I also moved some code around within the file so that the channel code and call code was cleanly separated.

I also did some additional clean-up, including:

- changed function names such that members of the filter struct are prefixed with `cc_` but nothing else is
- changed type of second parameter of pick_subchannel() function from `void *` to `grpc_call_element *`
- eliminated the pick_subchannel member of the call_data struct, since we can now directly call the local static function without requiring a layer of indirection